yt_idefix/data_structures.py

```diff
@@ -443,76 +530,36 @@ class PlutoVtkDataset(IdefixVtkDataset):
-        unit_system: Literal["cgs", "mks", "code"] = "cgs",
-        default_species_fields: Literal["neutral", "ionized"] | None = None,
-        # from here, frontend-specific arguments
-        geometry: Literal["cartesian", "spherical", "cylindrical", "polar"]
-        | None = None,
-        inifile: str | os.PathLike[str] | None = None,
-        definitions_header: str | None = None,
-    ):
-        self._definitions_header: str | None
-        if definitions_header is not None:
-            self._definitions_header = os.fspath(definitions_header)
-        else:
-            self._definitions_header = None
-
-        super().__init__(
-            filename,
-            dataset_type=dataset_type,
-            units_override=units_override,
-            unit_system=unit_system,
-            geometry=geometry,
-            inifile=inifile,
-            default_species_fields=default_species_fields,
-        )
+    _header_type = Header.C
+    _default_inifile = "pluto.ini"
 
     def _parse_parameter_file(self):
-        self._parse_header_file()
         super()._parse_parameter_file()
+
+        # TODO: this function could now be inilined
         self._get_time()
 
-    def _parse_header_file(self):
+    def _parse_definitions_header(self) -> None:
         """Read some metadata from header file 'definitions.h'."""
+        if not self._definitions_header:
+            return
+
         geom_regexp = re.compile(r"^\s*#define\s+GEOMETRY\s+([A-Z]+)")
         unit_regexp = re.compile(r"^\s*#define\s+UNIT_(\w+)\s+(\S+)")
         constexpr = re.compile(r"CONST_\w+")
 
-        # definitions.h is presumed to be along with data file
-        if self._definitions_header is None:
-            self._definitions_header = os.path.join(self.directory, "definitions.h")
-        elif not os.path.isfile(self._definitions_header):
-            raise FileNotFoundError(f"No such file {self._definitions_header!r}")
-
-        if os.path.isfile(self._definitions_header):
-            with open(self._definitions_header) as fh:
-                body = fh.read()
-            lines = C_io.strip_comments(body).split("\n")
-
-            for line in lines:
-                geom_match = re.fullmatch(geom_regexp, line)
-                if geom_match is not None:
-                    self.parameters["geometry"] = geom_match.group(1).lower()
-                    continue
-
-                unit_match = re.fullmatch(unit_regexp, line)
-                if unit_match is not None:
-                    unit = unit_match.group(1).lower() + "_unit"
-                    expr = unit_match.group(2)
-                    expr = re.sub(constexpr, self._get_constants, expr)
-                    self.parameters[unit] = eval(expr)
-        else:
-            warnings.warn(
-                f"Header file {self._definitions_header} couldn't be found. "
-                "The code units are set to be 1.0 in cgs by default."
-            )
```

Comment:
I think this warning is better to be reserved to remind users that they maybe forget the header file since it's common to define the code units in Pluto.

Reply:
I'm removing it because with my patch, defining a non-existent header file will result in a FileNotFoundError on `open`, so the warning is unreachable. Does it make sense ?

Reply:
How about showing this warning after checking the `self._definitions_header` is None, and then return?

Reply:
does https://github.com/neutrinoceros/yt_idefix/pull/169/commits/b3857295f8019bbfd11177fcae5fc662ad0064e5 work for you ?

Reply:
Yes, that's great!